packages/pluggableWidgets/switch-web/src/components/Switch.tsx

```diff
@@ -0,0 +1,46 @@
+import { MouseEvent, KeyboardEvent, createElement } from "react";
+import classNames from "classnames";
+import { Alert } from "@mendix/piw-utils-internal";
+import { SwitchContainerProps } from "../../typings/SwitchProps";
+
+export interface SwitchProps extends Pick<SwitchContainerProps, "id" | "class" | "style" | "tabIndex" | "deviceStyle"> {
+    onClick?: (event: MouseEvent<HTMLDivElement>) => void;
+    onKeyDown?: (event: KeyboardEvent<HTMLDivElement>) => void;
+    isChecked: boolean;
+    editable: boolean;
+    validation: string | undefined;
+}
+
+export default function Switch(props: SwitchProps) {
+    return (
+        <div className={classNames("widget-switch", props.class, props.deviceStyle)} style={props.style}>
+            <input
+                checked={props.isChecked}
+                className={classNames("widget-switch-checkbox", { enabled: props.editable })}
+                readOnly
+                type="checkbox"
+            />
+            <div
+                className={classNames("widget-switch-btn-wrapper", "widget-switch-btn-wrapper-default", {
+                    checked: props.isChecked,
+                    disabled: !props.editable,
+                    "un-checked": !props.isChecked
+                })}
+                onClick={props.onClick}
+                onKeyDown={props.onKeyDown}
+                tabIndex={props.tabIndex ?? 0}
+                role="checkbox"
+                aria-checked={props.isChecked}
+                aria-labelledby={props.id}
```

Comment:
I don't know if I understand you correctly. 

IIUC, the label system property will handle rendering the label widget client-side, then for accessibility purposes this `id` is passed here, so assistive tech can "link" the two elements.

Reply:
They're they same value, which is the `id` of the widget, right? With a systemProp `Label` configured, the widget receives the following props:

```
{
    "name": "switch1",
    "deviceStyle": "auto",
    "booleanAttribute": {
        "status": "available",
        "value": false,
        "displayValue": "No",
        "readOnly": false,
        "formatter": {},
        "universe": [
            true,
            false
        ]
    },
    "id": "0.MyFirstModule.Home_Web.switch1.24_cor_4"
}
```

Reply:
And the label element is e.g.

`<label class="control-label col-sm-6" for="0.MyFirstModule.Home_Web.switch1.24_cor_4" id="0.MyFirstModule.Home_Web.switch1.24_cor_4-label">Switch</label>`

Reply:
Aha, I see. The aria-labelledby prop would need the suffix "-label" though in this case. However, aria-labelledby isn't necessary, because we have the `for` attribute already on label: https://stackoverflow.com/questions/11152323/the-purpose-of-using-aria-labelledby-on-already-labeled-input-elements

Reply:
@IIsaku what do you mean the attribute/prop needs a suffix? The prop/attribute is `aria-labelledby` not `aria-labelledby-label`. 

I did some reading, while the label's `for` and the div's `id` might be sufficient for assistive  technologies, there's no harm in keeping the `aria-labelledby` attribute on the div. There might be some assitive tech that favours this `aria-labelledby` attribute, it's hard to say. I'm leaning towards keeping it for this reason, plus backwards compatibility. 

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-labelledby_attribute

Reply:
Good point, let's keep both.

The id of the label is "0.MyFirstModule.Home_Web.switch1.24_cor_4-label", but props.id contains just "0.MyFirstModule.Home_Web.switch1.24_cor_4", so you would have to apply "-label" as a suffix.